arch/arc/core/reset.S

```diff
@@ -121,6 +121,36 @@ invalidate_dcache:
 
 done_cache_invalidate:
 
+/*
+ * ARC Custom Initialization
+ * For scenarios where board hardware is not reset between tests,
+ * Some settings need to be reset by software.
+ */
+#if defined(CONFIG_ARC_CUSTOM_INIT)
+	/* Software Reset MPU (v3) registers */
+#if CONFIG_ARC_MPU_VER == 3
+	/* Set default reset value to _ARC_V2_MPU_EN */
+	mov_s r1, ARC_MPU_EN_RESET_VALUE
+	sr r1, [_ARC_V2_MPU_EN]
+	/* Get MPU region numbers */
+	lr r3, [_ARC_V2_MPU_BUILD]
+	lsr_s r3, r3, 8
+	and r3, r3, 0xff
+	mov_s r1, 0
+	mov_s r2, 0
+	/* Reset all MPU regions by iterating index */
+mpu_software_reset:
```

Comment:
Here as well that `mpu_software_reset` label is a bit confusing.
Maybe something more functional like `mpu_regions_reset`?